Makefile

```diff
@@ -15,28 +15,37 @@
 # specific language governing permissions and limitations
 # under the License.
 
-install-poetry:
-	pip install poetry==1.8.3
 
-install-dependencies:
-	poetry install -E pyarrow -E hive -E s3fs -E glue -E adlfs -E duckdb -E ray -E sql-postgres -E gcsfs -E sql-sqlite -E daft
+help:  ## Display this help
+	@awk 'BEGIN {FS = ":.*##"; printf "\nUsage:\n  make \033[36m\033[0m\n"} /^[a-zA-Z_-]+:.*?##/ { printf "  \033[36m%-15s\033[0m %s\n", $$1, $$2 } /^##@/ { printf "\n\033[1m%s\033[0m\n", substr($$0, 5) } ' $(MAKEFILE_LIST)
```

Comment:
nit:

```suggestion
	@awk 'BEGIN {FS = ":.*##"; printf "\nUsage:\n  make \033[36m\033[0m\n"} /^[a-zA-Z_-]+:.*?##/ { printf "  \033[36m%-20s\033[0m %s\n", $$1, $$2 } /^##@/ { printf "\n\033[1m%s\033[0m\n", substr($$0, 5) } ' $(MAKEFILE_LIST)
```

so that we can vertically align the starting position of the help descriptions from

```
Usage:
  make 
  help             Display this help
  install-poetry   Install poetry if the user has not done that yet.
  install-dependencies  Install dependencies including dev and all extras
  check-license    Check license headers
  lint             lint
  test             Run all unit tests, can add arguments with PYTEST_ARGS="-vv"
  test-integration  run all integration tests with potentially arguments like PYTEST_ARGS="-vv"
  test-adlfs       Run tests marked with adlfs, can add arguments with PYTEST_ARGS="-vv"
  test-gcs         Run tests marked with gcs, can add arguments with PYTEST_ARGS="-vv"
  test-coverage    Run all tests with coverage including unit and integration tests
  clean            clean up the project Python working environment
```

to:

```
Usage:
  make 
  help                  Display this help
  install-poetry        Install poetry if the user has not done that yet.
  install-dependencies  Install dependencies including dev and all extras
  check-license         Check license headers
  lint                  lint
  test                  Run all unit tests, can add arguments with PYTEST_ARGS="-vv"
  test-integration      run all integration tests with potentially arguments like PYTEST_ARGS="-vv"
  test-adlfs            Run tests marked with adlfs, can add arguments with PYTEST_ARGS="-vv"
  test-gcs              Run tests marked with gcs, can add arguments with PYTEST_ARGS="-vv"
  test-coverage         Run all tests with coverage including unit and integration tests
  clean                 clean up the project Python working environment
```